src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -6564,6 +6564,33 @@ components:
     ## ⚠️ NOTE: ALL SNAPSHOT API MODELS ARE SUBJECT TO CHANGE
     ##############################################################################
     SnapshotBuilderSettings:
+      type: object
+      description: >
+        ⚠️ The configuration describing how select criteria for issuing a snapshot builder request is generated
+      required:
+        - domainOptions
+        - programDataOptions
+        - featureValueGroups
+      properties:
+        selectableDomains:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderDomain'
+        selectableProgramData:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderProgramData'
+        featureValueGroups:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderFeatureValueGroup'
+        prepackagedDatasetConceptSets:
+          type: array
+          items:
+            $ref:
+              '#/components/schemas/SnapshotBuilderDatasetConceptSets'
+
+    SnapshotBuilderSettingsOptions:
```

Comment:
I realized that the objects we want to store in our DB don't quite match what we want to give the user, because some of the data is going to be derived, but we probably don't care as much about showing how it is derived to researchers. So I split these into two different objects. These were the best names I could come up with.